content/courses/mobile/solana-mobile-dapps-with-expo.md

```diff
@@ -460,18 +469,42 @@ export function MainScreen() {
 }
 ```
 
+Next, create file called `polyfills.ts` for react-native to work with all solana
+dependencies
+
+```typescript
+import { getRandomValues as expoCryptoGetRandomValues } from "expo-crypto";
+import { Buffer } from "buffer";
+
+global.Buffer = Buffer;
+
+// getRandomValues polyfill
+class Crypto {
+  getRandomValues = expoCryptoGetRandomValues;
+}
+
+const webCrypto = typeof crypto !== "undefined" ? crypto : new Crypto();
+
+(() => {
+  if (typeof crypto === "undefined") {
+    Object.defineProperty(window, "crypto", {
+      configurable: true,
+      enumerable: true,
+      get: () => webCrypto,
+    });
+  }
+})();
```

Comment:
Why the IIFE?

Reply:
I paste it from the [solana mobile docs](https://docs.solanamobile.com/react-native/expo)
![Screenshot from 2024-09-13 12-07-18](https://github.com/user-attachments/assets/7a5ec35a-25a2-4eea-b886-83b7e56e5649)

Reply:
OK. Run the code, delete the IIFI, does it still work?

Reply:
Also you have `typeof crypto === "undefined")` and `typeof crypto !== "undefined")` which is redundant. Try:

```
const hasInbuiltWebCrypto = typeof window.crypto === "undefined";

const webCrypto = hasInbuiltWebCrypto ? window,crypto : new Crypto();

if ( ! hasInbuiltWebCrypto ) {
  Object.defineProperty(window, "crypto", {
   configurable: true,
    enumerable: true,
    get: () => webCrypto,
  });
}
```